### PR TITLE
fix race in eventer startup resulting in unattended events

### PR DESCRIPTION
**Pull request by postwait**

### CODE REVIEW

File: src/eventer/eventer_impl.c [hunk 1]
```diff
@@ -412,13 +412,21 @@ mtev_boolean eventer_watchdog_timeout_timeval(struct timeval *dur) {
   return mtev_watchdog_get_timeout_timeval(t->hb, dur);
 }
 
+static void *eventer_get_spec_for_thread(struct eventer_impl_data *t) {
+  if(t->spec == NULL) {
```
Comment (pjulien) [inline src/eventer/eventer_impl.c:416]: This still races.  Need at the very least an acquire/release fence for getting and setting the value of the pointer.  This is the cheapest fence so any other stronger fence will also do.

Comment (pamaddox) [reply src/eventer/eventer_impl.c:416]: We check the value again inside the lock - we see if null, acquire lock, then see if null to make sure it didn't get set in that brief window.

Comment (pjulien) [reply src/eventer/eventer_impl.c:416]: You're incorrect, this still races.

Comment (pjulien) [reply src/eventer/eventer_impl.c:416]: The pointer must be read and stored in a local variable using this [ck_pr_fence_acquire](http://concurrencykit.org/doc/ck_pr_fence_acquire.html)

Then it must be transferred from the local variable to the field using [ck_pr_fence_release](http://concurrencykit.org/doc/ck_pr_fence_release.html)

**Pull request merged**